include/zephyr/dt-bindings/pinctrl/quicklogic-eos-s3-pinctrl.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_PINCTRL_QUICKLOGIC_EOS_S3_PINCTRL_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_PINCTRL_QUICKLOGIC_EOS_S3_PINCTRL_H_
+
+#include <zephyr/sys/util_macro.h>
+
+#define IO_MUX_REG_MAX_OFFSET	107
+#define IO_MUX_MAX_PAD_NR	45
+
+#define EOS_S3_UART_TXD_SEL_PAD44	(0x00000003)
+#define EOS_S3_PAD_E_4MA		BIT(8)
+#define EOS_S3_UART_RXD_SEL_PAD45	BIT(2)
+#define EOS_S3_PAD_OEN_DISABLE		BIT(5)
+#define EOS_S3_PAD_OEN_NORMAL		(0x00000000)
+#define EOS_S3_PAD_REN_ENABLE		BIT(11)
+#define EOS_S3_PAD_P_Z			(0x00000000)
+#define EOS_S3_PAD_SMT_DISABLE		(0x00000000)
+#define EOS_S3_PAD_REN_DISABLE		(0x00000000)
+#define EOS_S3_PAD_SR_SLOW		(0x00000000)
+#define EOS_S3_PAD_CTRL_SEL_FPGA	BIT(4)
+
+#define QUICKLOGIC_EOS_S3_PINMUX_USB_PU_CTRL_PAD	23
+#define QUICKLOGIC_EOS_S3_PINMUX_USB_DN_PAD		28
+#define QUICKLOGIC_EOS_S3_PINMUX_USB_DP_PAD		31
+#define QUICKLOGIC_EOS_S3_PINMUX_UART_TX_PAD		44
+#define QUICKLOGIC_EOS_S3_PINMUX_UART_RX_PAD		45
+
+#define UART_RX_SEL	EOS_S3_UART_RXD_SEL_PAD45
+
+#define QUICKLOGIC_EOS_S3_PINMUX_FUNC_SEL_NONE 0
+#define QUICKLOGIC_EOS_S3_PINMUX_FUNC_SEL_UART_RX	(77 << 13)
+
+#define QUICKLOGIC_EOS_S3_PINMUX_UART_RX_PAD_CFG \
+	(QUICKLOGIC_EOS_S3_PINMUX_FUNC_SEL_UART_RX \
+	 | EOS_S3_UART_RXD_SEL_PAD45 | EOS_S3_PAD_OEN_DISABLE \
+	 | EOS_S3_PAD_E_4MA | EOS_S3_PAD_REN_ENABLE)
+#define QUICKLOGIC_EOS_S3_PINMUX_UART_TX_PAD_CFG \
+	(EOS_S3_UART_TXD_SEL_PAD44 | EOS_S3_PAD_E_4MA)
+
+#define QUICKLOGIC_EOS_S3_PINMUX_USB_PAD_CFG (EOS_S3_PAD_E_4MA | EOS_S3_PAD_P_Z \
+					      | EOS_S3_PAD_OEN_NORMAL | EOS_S3_PAD_SMT_DISABLE \
+					      | EOS_S3_PAD_REN_DISABLE \
+					      | EOS_S3_PAD_SR_SLOW | EOS_S3_PAD_CTRL_SEL_FPGA)
```

Comment:
please use standard properties, 4MA looks like a drive strength...

Reply:
I've reworked the driver so that it utilizes properties from the DTS. Could you take a look at the current implementation? Thanks!